packages/other/src/DragList.ts

```diff
@@ -0,0 +1,197 @@
+import { HTMLWidget, mouse as d3Mouse, select as d3Select, selectAll as d3SelectAll, Widget } from "@hpcc-js/common";
+
+export class DragList extends HTMLWidget {
+
+    _container;
+    _containerTop;
+    _dragMemory: any = {};
+
+    constructor() {
+        super();
+    }
+
+    removeItem(node: HTMLElement) {
+        const item = node.parentElement.parentElement;
+        const order = Number(item.getAttribute("data-order"));
+        this.widgets().splice(order, 1);
+        this.render();
+    }
+
+    applySortOrder() {
+        const order = d3SelectAll(".drag-list-item").nodes()
+            .map((elm: any) => Number(elm.getAttribute("data-order")))
+            ;
+
+        const orderedWidgetArr = [...this.widgets()];
+
+        order.forEach((val, i) => { orderedWidgetArr[val] = this.widgets()[i]; });
+
+        this.widgets(orderedWidgetArr);
+        this.render();
+    }
+
+    enter(domNode, element) {
+        super.enter(domNode, element);
+
+        const context = this;
+
+        this._container = element.append("div")
+            .attr("class", "drag-list-container")
+            .style("position", "relative")
+            .style("overflow-x", "hidden")
+            .style("overflow-y", "auto")
+            ;
+
+        element.on("mousemove", function() {
+            if (context._dragMemory.element) {
+                const coordinates = d3Mouse(this);
+                const y = coordinates[1];
+                const parentScrollTop = context.element().node().parentElement.parentElement.scrollTop;
```

Comment:
`context.element()` should be the same as `element` and `context.element().node()` should be the same as `domNode `?